### DISCUSSION TITLE
LiveReload is not working

### DISCUSSION THREAD

**Discussion by H-Ashrafi** [statiqdev/Discussions#183] [General]
Hi I use Hugo most of the time but decided to give Statiq a try,. Unfortunately LiveReload is not working for me, I don’t know if it has something to do with my machine or it is a bug in Statiq.
I installed the Statiq.web from Nugget.

**Comment by daveaglick** [comment]
A couple questions to see if we can figure out what's going on...

Does the built-in web server work? I.e. if you do `dotnet run -- preview` can you then go to http://localhost:5080 and see the site?

If the preview web server is working, do you see execution kicking off in the console when you make file changes? The console should produce a lot of output when you change and save a file.

If you see execution going on each file save, then the watcher is working. Do you see traffic like the following in the browser devtools in that case:

![image](https://user-images.githubusercontent.com/1020407/209998521-d28692e4-6f08-4fce-a590-850e69354943.png)

Live reload works by establishing a websocket connection between the browser and the preview web server. When Statiq rebuilds the site on file changes it "pushed" a websocket message to the client that tells the client to reload the current page. This is managed on the client by injecting some JavaScript that you should be able to see as well:

![image](https://user-images.githubusercontent.com/1020407/209999079-78aaeea7-6dc5-4862-928c-f48f1c5c073a.png)

**Reply by daveaglick** [reply]
One more question though: is this on a very simple generation with just a Markdown file and no layouts or theme or anything? It's possible the live reload script isn't being injected in those cases due to the lack of `<html>`, etc. since Markdown doesn't usually produce the "boilerplate" HTML around the rendered markup.

I'm looking into this particular case now, though no idea if that's what's going on in your case or not.

**Comment by H-Ashrafi** [comment]
@daveaglick, 
- The page builds and I can see it in the console outputs and also when I manually reload the page, I can see the changes. the problem is that the JavaScript that you mentioned does not get injected into the browser (neither into Edge nor into FireFox) and LiveReload doesn't work.
- However, when I use Hugo, this does not happen. It can inject the <script> tag so I get LiveReload.
- I followed the QuickStart guide in the [Statiq.web ](https://www.statiq.dev/web) so I only created one index.md and tested everything there.
- To double check everything with another tool I used [Azure swa cli](https://azure.github.io/static-web-apps-cli/docs/use/install/) and had it pointing to the output folder that Statiq creates then in another terminal ran the `dotnet run -- preview` command so statiq would automatically output content but even SWA cannot inject that JS code into the browser.
![statiq](https://user-images.githubusercontent.com/31099988/210030009-1abac9bf-8304-4d13-8761-922b18364104.jpg)

**Reply by daveaglick** [reply]
> I followed the QuickStart guide in the Statiq.web so I only created one index.md and tested everything there.

Okay, that makes sense. It's the issue I was working earlier today (the injection code was looking for a closing `</body>` tag that doesn't exist when rendering only Markdown). I think it's fixed in the latest version released a few hours ago. When you get a chance can you update and let me know if that resolves the problem?

**Reply by H-Ashrafi** [reply]
Yes that fixed it, thank you.
